x-pack/plugins/fleet/public/components/platform_selector.tsx

```diff
@@ -37,19 +43,37 @@ const CommandCode = styled.pre({
   overflow: 'auto',
 });
 
-const K8S_COMMAND = `kubectl apply -f elastic-agent-managed-kubernetes.yaml`;
-
 export const PlatformSelector: React.FunctionComponent<Props> = ({
   linuxCommand,
   macCommand,
   windowsCommand,
   linuxDebCommand,
   linuxRpmCommand,
+  k8sCommand,
   isK8s,
+  isManaged,
+  enrollToken,
+  isFleet,
   fullCopyButton,
   onCopy,
 }) => {
   const { platform, setPlatform } = usePlatform();
+
+  useEffect(() => {
+    setPlatform(isK8s ? 'kubernetes' : 'linux');
+  }, [isK8s, isManaged, setPlatform]);
+
+  // In case of fleet server installation remove Kubernetes as platform option
+  const REDUCED_PLATFORM_OPTIONS = [...PLATFORM_OPTIONS];
+  const indexOfK8s = REDUCED_PLATFORM_OPTIONS.findIndex((object) => {
+    return object.id === 'kubernetes';
+  });
+  if (indexOfK8s !== -1) {
+    REDUCED_PLATFORM_OPTIONS.splice(indexOfK8s, 1);
+  }
```

Comment:
I have a suggestion to avoid all of this, see my comment below on `use_platform`